lib/internal/async_hooks.js

```diff
@@ -178,13 +178,11 @@ function fatalError(e) {
 
 function lookupPublicResource(resource) {
   if (typeof resource !== 'object' || resource === null) return resource;
-
-  const publicResource = resource[owner_symbol];
-
-  if (publicResource != null) {
+  // TODO(addaleax): Merge this with owner_symbol and use it across all
+  // AsyncWrap instances.
+  const publicResource = resource[resource_symbol];
+  if (publicResource !== undefined)
```

Comment:
```suggestion
  if (publicResource != null)
```

Reply:
@Flarna why though? Could `publicResource` be `null` in any case that I'm overlooking?

Reply:
Looked again. Seems I got confused during the review. It's fine as it is.